pkg/operator/status/status_controller.go

```diff
@@ -78,6 +79,20 @@ func NewClusterOperatorStatusController(
 	}
 }
 
+// WithRelatedObjectsFunc allows the set of related objects to be dynamically
+// determined.
+//
+// If the function returns nil, the set of related objects is carried over
+// from the existing ClusterOperator. This is useful if the set of related
+// objects is not yet known, perhaps because a control loop hasn't yet
+// executed. To set no related objects, return empty list.
```

Comment:
strange signature for this behaviour. I would expect at least c.relatedObject to be applied, always.

Reply:
Right, but then we could flap RelatedObjects.

Here's the motivating example: on restart, the CNO may wish to set a Degraded status condition immediately, but not wish to touch RelatedObjects.

Returning nil signifies "have some inertia - don't touch anything". One could, I suppose, ensure that the static `c.relatedObjects` is in the ClusterOperator object.